packages/repository/test/unit/repository-mixin/repository-mixin.test.ts

```diff
@@ -0,0 +1,125 @@
+
+  it('binds repository from constructor', () => {
+    const myApp = new appWithRepoMixin({
+      repositories: [NoteRepo],
+    });
+
+    const boundRepositories = myApp.find('repositories.*').map(b => b.key);
+    expect(boundRepositories).to.containEql('repositories.NoteRepo');
+
+    const repoInstance = myApp.getSync('repositories.NoteRepo');
+    expect(repoInstance).to.be.instanceOf(NoteRepo);
+  });
+
+  it('binds repository from app.repository()', () => {
+    const myApp = new appWithRepoMixin();
+
+    const boundReposBefore = myApp.find('repositories.*').map(b => b.key);
+    expect(boundReposBefore).to.be.empty();
+
+    myApp.repository(NoteRepo);
+
+    const boundReposAfter = myApp.find('repositories.*').map(b => b.key);
+    expect(boundReposAfter).to.containEql('repositories.NoteRepo');
+
+    const repoInstance = myApp.getSync('repositories.NoteRepo');
+    expect(repoInstance).to.be.instanceOf(NoteRepo);
+  });
+
+  it('binds user defined component without repository', () => {
+    class TestComponent {}
+
+    const myApp = new appWithRepoMixin({
+      components: [TestComponent],
+    });
+
+    const componentKeys = myApp.find('components.*').map(b => b.key);
+    expect(componentKeys).to.containEql('components.TestComponent');
+    const componentInstance = myApp.getSync('components.TestComponent');
+    expect(componentInstance).to.be.instanceOf(TestComponent);
+  });
+
+  it('binds user defined component with repository in constructor', () => {
+    class TestComponent {
+      repositories = [NoteRepo];
+    }
+
+    const myApp = new appWithRepoMixin({
+      components: [TestComponent],
+    });
+
+    const componentKeys = myApp.find('components.*').map(b => b.key);
+    expect(componentKeys).to.containEql('components.TestComponent');
+    const componentInstance = myApp.getSync('components.TestComponent');
+    expect(componentInstance).to.be.instanceOf(TestComponent);
+
+    const boundRepositories = myApp.find('repositories.*').map(b => b.key);
+    expect(boundRepositories).to.containEql('repositories.NoteRepo');
+    const repoInstance = myApp.getSync('repositories.NoteRepo');
+    expect(repoInstance).to.be.instanceOf(NoteRepo);
+  });
+
+  it('binds user defined component with repository from .component()', () => {
+    class TestComponent {
+      repositories = [NoteRepo];
+    }
+
+    const myApp = new appWithRepoMixin();
+
+    const boundComponentsBefore = myApp.find('components.*').map(b => b.key);
+    expect(boundComponentsBefore).to.be.empty();
+    const boundReposBefore = myApp.find('repositories.*').map(b => b.key);
+    expect(boundReposBefore).to.be.empty();
+
+    myApp.component(TestComponent);
+
+    const boundComponentsAfter = myApp.find('components.*').map(b => b.key);
+    expect(boundComponentsAfter).to.containEql('components.TestComponent');
+    const componentInstance = myApp.getSync('components.TestComponent');
+    expect(componentInstance).to.be.instanceOf(TestComponent);
```

Comment:
Ditto: I think these four checks are redundant (the previous test has already verified that a component is correctly loaded), and should be removed.

Reply:
I think this is a different case because the previous test checks to see if a component is bound when provided in the constructor whereas here it is provided via `app.component()`